### PR TITLE
Weight bug fixes

### PR DESCRIPTION
**Pull request by keymane**
Fix for https://github.com/OpenSRP/opensrp-client/issues/544 and https://github.com/OpenSRP/opensrp-client/issues/539

### CODE REVIEW

File: opensrp-path/src/main/java/org/ei/opensrp/path/viewComponents/WidgetFactory.java [hunk 1]
```diff
@@ -53,15 +48,15 @@ public View createTableRowForWeight(LayoutInflater inflater, ViewGroup container
...
         LinearLayout tableLayout = (LinearLayout) fragmentContainer.findViewById(R.id.weightvalues);
         tableLayout.removeAllViews();
-        ViewGroup.LayoutParams weightvaluesparams = tableLayout.getLayoutParams();
 
         int i = 0;
-        for (Map.Entry<String, String> entry : last_five_weight_map.entrySet()) {
-            View view = createTableRowForWeight(inflater, tableLayout, "" + entry.getKey(), "" + entry.getValue(), editenabled.get(i), listeners.get(i));
+        for (Map.Entry<Long, Pair<String, String>> entry : last_five_weight_map.entrySet()) {
+            Pair<String, String> pair = entry.getValue();
+            View view = createTableRowForWeight(inflater, tableLayout, "" + pair.first, "" + pair.second, editenabled.get(i), listeners.get(i));
```
Comment (pld) [inline opensrp-path/src/main/java/org/ei/opensrp/path/viewComponents/WidgetFactory.java]: know it's legacy, but why the `"" +`? Is that a janky cast?

File: opensrp-path/src/main/java/org/ei/opensrp/path/tabfragments/ChildUnderFiveFragment.java [hunk 2]
```diff
@@ -145,11 +144,11 @@ private void createWeightLayout(LinearLayout fragmentContainer, boolean editmode
                 }
             }
             if (!formattedAge.equalsIgnoreCase("0d")) {
-                weightmap.put(formattedAge, weightlist.get(i).getKg() + " kg");
+                weightmap.put(weight.getId(), Pair.create(formattedAge, weightlist.get(i).getKg() + " kg"));
```
Comment (pld) [inline opensrp-path/src/main/java/org/ei/opensrp/path/tabfragments/ChildUnderFiveFragment.java]: this and line 187 feel like they are begging for a `kilogramStringSuffix` helper or similar

File: opensrp-path/src/main/java/org/ei/opensrp/path/provider/ChildSmartClientsProvider.java [hunk 3]
```diff
@@ -140,7 +140,7 @@ public void getView(SmartRegisterClient client, View convertView) {
         Weight weight = weightRepository.findUnSyncedByEntityId(pc.entityId());
         if (weight != null) {
             TextView recordWeightText = (TextView) convertView.findViewById(R.id.record_weight_text);
-            recordWeightText.setText(weight.getKg().toString() + " kg");
+            recordWeightText.setText(String.format("%s kg", weight.getKg()));
```
Comment (pld) [inline opensrp-path/src/main/java/org/ei/opensrp/path/provider/ChildSmartClientsProvider.java]: see comment below

File: opensrp-path/src/main/java/org/ei/opensrp/path/viewComponents/WidgetFactory.java (hunk 1 above)
Comment (keymane) [inline opensrp-path/src/main/java/org/ei/opensrp/path/viewComponents/WidgetFactory.java]: Will remove them

**Pull request merged**